testutils/clustermanager/e2e-tests/gke.go

```diff
@@ -276,20 +262,21 @@ func (gc *GKECluster) Delete() error {
 		return gc.boskosOps.ReleaseGKEProject(gc.Project)
 	}
 
-	// NeedsCleanup is only true if running locally and cluster created by the
-	// process
-	if !gc.NeedsCleanup && !gc.Request.NeedsCleanup {
```

Comment:
So this leaves the responsibility of these conditions to caller?

Reply:
I'm changing the logic a bit:
Delete the cluster async is it's on Prow, but delete it sync if it's on local. I also added it in the PR description. Please let me know how you feel about it.

Reply:
So when a user runs this locally, it will always delete?

Reply:
Yes, it's the original logic...

Reply:
but can't user ask not to delete in `gc.Request`?

Reply:
The logic looks quite weird to me if user does not want to delete the cluster but explicitly call the delete function...   I think whether or not to delete the cluster should be determined by workflow by not here, e.g. by the `--skip-teardown` flag I added a while ago.

Reply:
That's ok with me